### PR TITLE
Modeling roundabouts

### PR DESCRIPTION
**Pull request by Andygol**
Guide for modeling circular road junctions

screenshot

### CODE REVIEW

File: mapping-for-navigation/_posts/0000-01-10-circular-junctions.md [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+[![][72941a36]][72941a36]
+
+<sup>_Mini roundabout in OSM_</sup>
+
+  [72941a36]: ../../images/intersection-modeling/roundabouts/roundabout_mapping_mini_roundabout.png "how to map mini-roundabout"
+
+Select the node on way crossing and add tags **[highway][1108080d]**=**[mini_roundabout][11be223c]**. Please note, that the key **[direction](https://wiki.openstreetmap.org/wiki/Key:direction)**=**\*** only needs to be specified if the traffic flow on the mini-roundabout is clockwise 🔁 as the default value for this key in conjunction with mini-roundabouts is anti-clockwise 🔄.
+
+Since the wiki suggests, we mark mini-roundabout as the place with traversable central point. But in some cases, we may find places with the presence of a traffic island on the central point of a roundabout. If there is the sign _roundabout_ tag it with **junction**=**roundabout** + **traffic_calming**=**island** tags on the node of the intersection, otherwise put only **traffic_calming**=**island** tag.
```
Comment (karenzshea) [inline mapping-for-navigation/_posts/0000-01-10-circular-junctions.md]: > As the wiki suggests, we mark mini-roundabouts at the place of the traversable centre point. In the case of roundabouts with a traffic island in the centre of the roundabout, if there is a sign _roundabout_ tag, tag it with ...

File: mapping-for-navigation/_posts/0000-01-10-circular-junctions.md [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+![][7c512a18]
+
+<sup>_Small roundabout with traffic island by **[Elaine Nelson](https://www.flickr.com/photos/epersonae/)** - <https://flic.kr/p/8n4kXv>_</sup>
+<!-- |<sup>_Small roundabout with traffic island by **[Matt Steele](https://www.flickr.com/photos/mattaudio/)** - <https://flic.kr/p/su9dcz>_</sup>| -->
+
+  [7c512a18]: https://c1.staticflickr.com/5/4153/4832796263_1277072e82_b.jpg "small roundabout"
+  <!-- [fbdc0d39]: https://c2.staticflickr.com/8/7774/17383739791_0a1b0dfb76_b.jpg "roundabout with a traffic island" -->
+
+Mini-roundabout always need to be represented by **node**. Another way is to draw the roundabout as circular line in the direction the traffic flows and tag it as  **highway=residential\|unclassified** + **oneway**=**yes** and **junction**=**roundabout**.
```
Comment (karenzshea) [inline mapping-for-navigation/_posts/0000-01-10-circular-junctions.md]: Mini-roundabout --> mini-roundabouts

> Mini-roundabouts always need to be represented by a **node**. Alternatively, one can draw the roundabout as a circular line ...

File: mapping-for-navigation/_posts/0000-01-10-circular-junctions.md [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+
+**Common Misconceptions**
+
+What is not a mini roundabout?
```
Comment (karenzshea) [inline mapping-for-navigation/_posts/0000-01-10-circular-junctions.md]: Consistent hyphenation, `mini-roundabout`?

File: mapping-for-navigation/_posts/0000-01-10-circular-junctions.md [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+
+-   Traffic calming island in the center of a junction - **[traffic_calming](https://wiki.openstreetmap.org/wiki/Key:traffic_calming)**=**[island](https://wiki.openstreetmap.org/wiki/Tag:traffic_calming%3Disland)**
+-   Turning circle - this is a circular feature with a traversable center point, which allows vehicles to turn around -  **[highway][1108080d]**=**[turning_circle][496a5a09]**
+-   Turning loop has the same function as Turning circle but with none-traversable center point -  **[highway][1108080d]**=**[turning_loop][e1236adf]**
```
Comment (karenzshea) [inline mapping-for-navigation/_posts/0000-01-10-circular-junctions.md]: > Turning loops have the same function as turning circles but with non-traversable center points

File: mapping-for-navigation/_posts/0000-01-10-circular-junctions.md [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+
+[bca8053e]: https://wiki.openstreetmap.org/wiki/Tag:junction%3Droundabout
+
+[f14df547]: https://wiki.openstreetmap.org/wiki/Key:junction
+
+See examples of mini-roundabouts in the [OpenStreetMap Wiki](https://wiki.openstreetmap.org/wiki/Tag:highway%3Dmini_roundabout#Mini-roundabouts).
+
+### Roundabout
+
+A [roundabout](https://en.wikipedia.org/wiki/Roundabout) is a generally circular road [junction](https://wiki.openstreetmap.org/wiki/Junctions) where the traffic on the roundabout has right of way and tagged with **[junction][f14df547]**=**[roundabout](https://wiki.openstreetmap.org/wiki/Tag:junction%3Droundabout)** tag. This tag is used only on road intersections where traffic on the roundabout has right of way as in the absence of traffic signals in these intersections.
```
Comment (karenzshea) [inline mapping-for-navigation/_posts/0000-01-10-circular-junctions.md]: > A [roundabout](https://en.wikipedia.org/wiki/Roundabout) is a generally circular road [junction](https://wiki.openstreetmap.org/wiki/Junctions) where the traffic on the roundabout has right of way. They are tagged with **[junction][f14df547]**=**[roundabout](https://wiki.openstreetmap.org/wiki/Tag:junction%3Droundabout)**.

I'd split up the first sentence for clarity.

Comment (karenzshea) [reply mapping-for-navigation/_posts/0000-01-10-circular-junctions.md]: > This tag is used only on road intersections where traffic on the roundabout has right of way in the absence of a traffic signal.

File: mapping-for-navigation/_posts/0000-01-10-circular-junctions.md [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+
+[![roundabout][108eb915]][108eb915]
+<sup>_Take a look at a [wireframe view][c134f465] of this roundabout to get some details from the imagery_</sup>
+
+  [108eb915]: ../../images/intersection-modeling/roundabouts/roundabout.png "roundabout"
+  [c134f465]: ../../images/intersection-modeling/roundabouts/roundabout.png "wireframe view"
+
+**How to map**
+
+To indicate a roundabout — draw a closed line in the direction the traffic flows , the same as you do it during the road tracing and tag it with the **[junction][f14df547]**=**[roundabout](https://wiki.openstreetmap.org/wiki/Tag:junction%3Droundabout)** tag and appropriate **`highway=*`** tag value.
```
Comment (karenzshea) [inline mapping-for-navigation/_posts/0000-01-10-circular-junctions.md]: > draw a closed line in the direction of the traffic flow, just as one would while tracing a road and tag it with ...

File: mapping-for-navigation/_posts/0000-01-10-circular-junctions.md [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+
+Roundabout is implied to be one way road but you may specify this explicitly by adding **[oneway](https://wiki.openstreetmap.org/wiki/Key:oneway)**=**[yes](https://wiki.openstreetmap.org/wiki/Tag:oneway%3Dyes)** tag. Additionally the number of **[lanes](https://wiki.openstreetmap.org/wiki/Key:lanes)**=**`*`** in the circle (typically 2, where long vehicles will need to use both) can also be tagged.
```
Comment (karenzshea) [inline mapping-for-navigation/_posts/0000-01-10-circular-junctions.md]: > Roundabouts are implied one-way roads, but ...

File: mapping-for-navigation/_posts/0000-01-10-circular-junctions.md [hunk 8]
```diff
@@ -0,0 +1,208 @@
...
+
+When two roads share the same exit/entrance node connect them before the roundabout.
+
+[![][8c17feb2]][8c17feb2]
+
+  [8c17feb2]: ../../images/intersection-modeling/roundabouts/roundabout_pinch_ways.png
+
+**Roundabout shapes**
+
+Roundabouts are not necessarily circular in shape: there does exist roundabouts that are 0-shaped (oval) or D-shaped (with a straight segment, notably those created on a square whose one side is along an avenue).
```
Comment (karenzshea) [inline mapping-for-navigation/_posts/0000-01-10-circular-junctions.md]: there does exist roundabouts that --> there are roundabouts that

just slightly less wordy

File: mapping-for-navigation/_posts/0000-01-10-circular-junctions.md [hunk 9]
```diff
@@ -0,0 +1,208 @@
...
+
+[![][c4d4f134]][c4d4f134]
+
+  [c4d4f134]: ../../images/intersection-modeling/roundabouts/roundabout_o_shape.png "O-shaped roundabout"
+
+See examples of roundabouts in the [OpenStreetMap Wiki](https://wiki.openstreetmap.org/wiki/Tag:junction%3Droundabout#Roundabouts).
+
+### Traffic circle
+
+A **traffic circle** is a type of intersection that directs both turning and through traffic onto a [one-way circular roadway](https://www.openstreetmap.org/user/daniel-j-h/diary/40101). Contrary to a roundabout, where entering traffic always yields to traffic already in the circle and merges in directly, the entrances to traffic circles are three-way intersections either controlled by stop signs, traffic signals, or not formally controlled.
```
Comment (karenzshea) [inline mapping-for-navigation/_posts/0000-01-10-circular-junctions.md]: > the entrances to traffic circles are three-way intersections often controlled either by stop signs or traffic signals, though not in all cases.

File: mapping-for-navigation/_posts/0000-01-10-circular-junctions.md [hunk 10]
```diff
@@ -0,0 +1,208 @@
...
+
+The size of circular junctions and rotaries can vary from small to very large with several lanes, however generally **<u>the rotary or circular junction should not have any road network inside itself</u>**, this means circular ring roads around a city are not valid circular junctions.
+
+To designate such case of circular junction use tag **[junction][f14df547]**=**[circular][d11d94d4]**.
+
+  [d11d94d4]: https://wiki.openstreetmap.org/wiki/Tag:junction%3Dcircular
+
+**How to map**
+
+To add such a junction, use the same approach as for drawing roundabouts. Draw a circle line and tag it accordingly to the class of the roads that join it. Mark the appropriate nodes with traffic lights and traffic signs that control the priority of the movement of traffic flows.
```
Comment (karenzshea) [inline mapping-for-navigation/_posts/0000-01-10-circular-junctions.md]: > Draw a circle line in the direction of the flow of traffic

Should that be clarified as it is in the roundabouts instructions?

File: mapping-for-navigation/_posts/0000-01-10-circular-junctions.md [hunk 11]
```diff
@@ -0,0 +1,208 @@
...
+
+If across a traffic circle or a roundabout go some routes that use different parts of that circles you may split the closed line into segments with the similar tagging and add each segment into respective route relation.
```
Comment (karenzshea) [inline mapping-for-navigation/_posts/0000-01-10-circular-junctions.md]: > If a traffic circle or roundabout is bisected by further roads that use different parts of the circle, you may split the closed line into segments with similar tagging and add each segment into respective route relation.

File: mapping-for-navigation/_posts/0000-01-10-circular-junctions.md [hunk 12]
```diff
@@ -0,0 +1,208 @@
...
+
+Turning circle - this is a circular feature with a traversable center point, which allows vehicles to turn around - **[highway][1108080d]**=**[turning_circle][496a5a09]**
+
+[![][86003e49]][86003e49]
+
+  [86003e49]: ../../images/intersection-modeling/roundabouts/roundabout_turning_circle.png
+
+### Turning loop
+
+Turning loop has the same mission as Turning circle but with none-traversable center point.
```
Comment (karenzshea) [inline mapping-for-navigation/_posts/0000-01-10-circular-junctions.md]: > Turning loops are similar to turning circles but with a non-traversable center point.

File: mapping-for-navigation/_posts/0000-01-10-circular-junctions.md [hunk 13]
```diff
@@ -0,0 +1,208 @@
...
+
+![Mini-roundabout](https://c2.staticflickr.com/6/5543/9459511782_dbfacf91f3_b.jpg)
+
+<sup>_A view of a mini roundabout from Church Lane with School Lane to the right and Church Street (A 623) to the left with a partial view of Saint Annes Church (Church of England) and War Memorial in the village of Baslow in Derbyshire. Photo Taken: 10/07/13_</sup>
+
+<sup>_by [Eamon Curry](https://www.flickr.com/photos/eamoncurry/) - <https://flic.kr/p/fpUthJ>_</sup>
+
+**Mini-roundabouts** can be a painted circle or a low dome but **must be fully traversable** by vehicles. Mini-roundabouts are sometimes grouped in pairs (a double mini-roundabout) or in "chains", simplifying navigation of otherwise awkward junctions. Vehicles are required to treat the painted circle as if it were a solid island and drive around it.
+
+![Mini-roundabout chaine](http://s0.geograph.org.uk/geophotos/03/06/29/3062935_47b91623.jpg)
```
Comment (daniel-j-h) [inline mapping-for-navigation/_posts/0000-01-10-circular-junctions.md]: This image shows a give-way sign. This is in contrast to the explanation above where we call out the right-of-way. If you want to keep this image I would specifically call out exceptions like this give-way sign here.

File: mapping-for-navigation/_posts/0000-01-10-circular-junctions.md (hunk 12 above)
Comment (Andygol) [reply mapping-for-navigation/_posts/0000-01-10-circular-junctions.md]: @karenzshea Thank you for your review and fix the wording. But one question: why this sentence in plural? What if we use singular here and above

> Turning loop is similar to the turning circle but with a non-traversable center point.

File: mapping-for-navigation/_posts/0000-01-10-circular-junctions.md (hunk 10 above)
Comment (Andygol) [reply mapping-for-navigation/_posts/0000-01-10-circular-junctions.md]: I've added anchor to link this to **how to** section of roundabout

```
**<a id="how-to-map-roundabout"></a>How to map**

To indicate a roundabout — draw a closed line in the direction the traffic flows,
…
To add such a junction, use the same approach as for [drawing roundabouts](#how-to-map-roundabout).
```

File: mapping-for-navigation/_posts/0000-01-10-circular-junctions.md (hunk 13 above)
Comment (Andygol) [reply mapping-for-navigation/_posts/0000-01-10-circular-junctions.md]: The main purpose of this image is to show `chained` mini-roundabouts when one comes next to another.

`give way` sign is applied to approaching traffic setting a higher priority to the traffic on the mini-roundabout.

**Comment by Andygol** [comment]
Recent version after fixing wording suggested by @karenzshea 

image

File: mapping-for-navigation/_posts/0000-01-10-circular-junctions.md [hunk 14]
```diff
@@ -0,0 +1,199 @@
...
+  [c134f465]: ../../images/intersection-modeling/roundabouts/roundabout_wireframe.png "wireframe view"
+
+**<a id="how-to-map-roundabout"></a>How to map a roundabout**
+
+To indicate a roundabout — draw a closed line in the direction of the traffic flow, which aligns to Satellite imagery just as one would while tracing a road. Tag the line with a **[junction][f14df547]**=**[roundabout](https://wiki.openstreetmap.org/wiki/Tag:junction%3Droundabout)**  tag and the appropriate **highway**=**\*** tag.
+
+Roundabouts are implied to be one-way roads, but you may specify this explicitly by adding **[oneway](https://wiki.openstreetmap.org/wiki/Key:oneway)**=**[yes](https://wiki.openstreetmap.org/wiki/Tag:oneway%3Dyes)** tag.
+
+Connect all the highways which meets the roundabout based on traffic flows. If a dual carriageway shares traffic with the roundabout, connect each its direction (one arriving, one departing) separately to the respective exit and entrance nodes on the roundabout. Don’t use the same node as entrance to and exit from a roundabout. Each road has to be connected with the roundabout in a separate node.
```
Comment (srividyacb) [inline mapping-for-navigation/_posts/0000-01-10-circular-junctions.md]: @Andygol This is a great point for roundabout modelling when dual carriages join the junction. Can you add a screenshot of what is recommended and what is not recommended here to make it clear with pictures ?

File: mapping-for-navigation/_posts/0000-01-10-circular-junctions.md [hunk 15]
```diff
@@ -0,0 +1,199 @@
...
+
+When two roads share the same exit/entrance node connect them before the roundabout and not at the roundabout.
+
+[![][8c17feb2]][8c17feb2]
+
+  [8c17feb2]: ../../images/intersection-modeling/roundabouts/roundabout_pinch_ways.png
+
+**Roundabout shapes**
+
+Roundabouts are not necessarily circular in shape: there are roundabouts that are oval-shaped or D-shaped (with a straight segment – often created on a square whose one side is along an avenue).
```
Comment (srividyacb) [inline mapping-for-navigation/_posts/0000-01-10-circular-junctions.md]: @Andygol Do we have example for D-shaped junctions ?

File: mapping-for-navigation/_posts/0000-01-10-circular-junctions.md (hunk 14 above)
Comment (Andygol) [reply mapping-for-navigation/_posts/0000-01-10-circular-junctions.md]: @srividyacb I've added screenshots https://github.com/mapbox/mapping/pull/352/commits/46926883fdaeb2b4a46b66197c8feab431356bf5

**Comment by Andygol** [comment]
This PR is ready for merging 🎉

**Pull request merged**